smarc_bt/src/mission_plan.py

```diff
@@ -473,6 +539,200 @@ def path_to_list(self, path_msg):
+                return intersections
+
+    def dubins_mission_planner(self, mission, num_points=2, inside_turn=True):
+        '''
+        Reads the waypoints from a MissionControl message and generates a sampled dubins
+        path between them.
+
+        :param mission: the original MissionControl message
+        :param num_points: the amount of waypoints to keep on each segment between waypoints
+        :param inside_turn: cut the waypoints corners
+        :return MissionControl.msg: new MissionControl message equal to the input one except for the waypoints attribute
+        '''
+
+        rospy.loginfo("Computing dubins path")
+
+        turn_radius = self.bb.get(bb_enums.DUBINS_TURNING_RADIUS)
+        int_radius = self.bb.get(bb_enums.DUBINS_INTERSECTION_RADIUS)
+        vehicle = self.bb.get(bb_enums.VEHICLE_STATE)
+        robot_name = self.bb.get(bb_enums.ROBOT_NAME)
+
+        ll_to_utm_serv = self._get_latlon_to_utm_service()
+
+        # Get the auv's current position to use as the intial point of the dubins path
+        utm_x_auv, utm_y_auv = vehicle.position_utm
+        auv_utm = np.array([utm_x_auv, utm_y_auv])
+
+        # Get x, y of waypoints from original mission lat/lon
+        points = []
+        for wp in mission.waypoints:
+            utm_x, utm_y = self.latlon_to_utm(wp.lat, wp.lon, wp.pose.pose.position.z, serv=ll_to_utm_serv, in_degrees=True)
+            points.append([utm_x, utm_y])
+        points_np = np.array(points)
+        points_with_auv = np.vstack((auv_utm, points_np))
+
+        if not inside_turn:
+            rospy.loginfo("Turning outside")
+            original_waypoints = points_with_auv
+        else:
+            rospy.loginfo("Turning inside")
+            waypoints = []
+            for i in range(len(points_with_auv)-1):
+                waypoints_i = self.circle_line_segment_intersection(circle_center=(points_with_auv[i, 0], points_with_auv[i, 1]), circle_radius=int_radius,
+                                                                    pt1=(points_with_auv[i, 0], points_with_auv[i, 1]), pt2=(points_with_auv[i+1, 0], points_with_auv[i+1, 1]))
+                waypoints.append(waypoints_i)
+                waypoints_j = self.circle_line_segment_intersection(circle_center=(points_with_auv[i+1, 0], points_with_auv[i+1, 1]), circle_radius=int_radius,
+                                                                    pt1=(points_with_auv[i, 0], points_with_auv[i, 1]), pt2=(points_with_auv[i+1, 0], points_with_auv[i+1, 1]))
+                waypoints.append(waypoints_j)
+
+            waypoints.append([(points_with_auv[-1, 0], points_with_auv[-1, 1])])
+            waypoints = [el for sublist in waypoints for el in sublist] # Flatten list of lists
+            waypoints_np = np.array(waypoints)
+            original_waypoints = waypoints_np.squeeze()  # Remove useless extra dimension
+
+        # Keep the other waypoints' parameters equal
+        mwp = mission.waypoints[0]
```

Comment:
doesnt this basically set all dubins-wp's parameters to be the same as the _first_ user-given wp's params? Each wp in a mission can have unique params, this discards them all.

if a plan is A->B->C, and gets dubins'ed to be A - x,y,z -> B -p,q,r-> C x,y,z should have the params of B and p,q,r should have the params of C